torch/csrc/lazy/backend/backend_device.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ostream>
+#include <memory>
+#include <string>
+
+#include <c10/macros/Export.h>
+#include <c10/util/Deprecated.h>
+
+namespace torch {
+namespace lazy {
+
+// Backend should extend it and define their own supported hardware types.
+struct TORCH_API BackendDeviceType {
+  int8_t type {0};
+
+  virtual ~BackendDeviceType() = default;
+  virtual std::string toString() const { return "Unknown"; }
+};
+
+class TORCH_API BackendDevice {
+ public:
+  BackendDevice();
+  BackendDevice(std::shared_ptr<BackendDeviceType>&& type, int ordinal);
+
+  int8_t type() const;
+  int64_t ordinal() const { return ordinal_;  }
+
+  bool operator==(const BackendDevice& other) const { return compare(other) == 0; }
+  bool operator!=(const BackendDevice& other) const { return compare(other) != 0; }
+  bool operator<(const BackendDevice& rhs) const { return compare(rhs) < 0; }
+
+  std::string toString() const;
+
+  // The string -> Device conversion should be handled by the backend interface.
+  C10_DEPRECATED explicit BackendDevice(const std::string& device_spec);
```

Comment:
am I supposed to delete this right away, once I merge my changes for backend interface? or we need to keep it around for a while longer?

Reply:
Good question. I think you can just delete this ctor during or after your merge PR.

Reply:
yea, backend can probably just add this if they want to(pt/xla will mostly add it on the extended Device class) .